pnc-rest/src/main/java/org/jboss/pnc/rest/provider/ProductProvider.java

```diff
@@ -24,8 +24,23 @@ public ProductProvider(ProductRepository productRepository) {
     public ProductProvider() {
     }
 
-    public List<ProductRest> getAll() {
-        return productRepository.findAll().stream().map(product -> new ProductRest(product)).collect(Collectors.toList());
+    @Override
+    public Function<? super Product, ? extends ProductRest> mapper() {
+        return product -> new ProductRest(product);
+    }
+
+    @Override
+    public String getDefaultSortingField() {
+        return "name";
+    }
+
+    public Object getAll(Integer pageIndex, Integer pageSize, String field, String sorting) {
+
+        if (noPaginationRequired(pageIndex, pageSize, field, sorting)) {
+            return productRepository.findAll().stream().map(mapper()).collect(Collectors.toList());
+        } else {
+            return transform(productRepository.findAll(buildPageRequest(pageIndex, pageSize, field, sorting)));
```

Comment:
Hmmm on the second thought, I see we are extracting everything from DB and apply pagination on those results. I think we should modify Datastore layer to accept `Pageable` objects. Just like it is done here: <http://docs.spring.io/spring-data/commons/docs/current/reference/html/#repositories.special-parameters>